### PR TITLE
mailheader: do not remove partial matches from header_list

### LINKED ISSUE #2641: transaction.remove_header removes too much
The following test plugin should remove the header `X-Test` but do not affect any other headers.

```js
exports.register = function () {
    const plugin = this;
    plugin.inherits('queue/quarantine')
    plugin.load_quarantine_ini();
    plugin.register_hook('queue',          'quarantine');
    plugin.register_hook('queue_outbound', 'quarantine');
}

exports.hook_rcpt = function (next, connection) {
    const plugin = this
    connection.notes.quarantine = true
    next(OK)
}

exports.hook_data_post = function (next, connection) {
    const plugin = this
    connection.transaction.remove_header('X-Test')
    next()
}

```

But when I call
```shell
swaks -s $HARAKA_HOST -p $HARAKA_PORT -t test@example.com --add-header "X-Test: remove-me" --add-header "X-Test-1: do-not-remove-me"
```

The `X-Test-1` header gets spuriously removed, too.

Looks like `_remove_more` in `mailheader.js` is the culprit:
https://github.com/haraka/Haraka/blob/4aea44d56834a1a4be78c56b646a6c266aa7d7b5/mailheader.js#L127-L135
(only comparing the header key where it should compare key + ':')

### PR DESCRIPTION
**Pull request by msimerson**
Fixes #2641

Changes proposed in this pull request:
- mailheader: add test for remove()
- mailheader: do not remove partial matches from header_list

Checklist:
- [ ] docs updated
- [x] tests updated
- [x] [Changes](https://github.com/haraka/Haraka/blob/master/Changes.md) updated

### CODE REVIEW

File: mailheader.js [hunk 1]
```diff
@@ -126,8 +125,8 @@ class Header {
 
     _remove_more (key) {
         const key_len = key.length;
-        for (let i=0,l=this.header_list.length; i < l; i++) {
-            if (this.header_list[i].substring(0, key_len).toLowerCase() === key) {
+        for (let i=0, l=this.header_list.length; i < l; i++) {
```
Comment (KingNoosh) [inline mailheader.js]: Can't we do `const header of this.header_list` here too?

File: mailheader.js [hunk 2]
```diff
@@ -126,8 +125,8 @@ class Header {
...
+            if (this.header_list[i].substring(0, key_len + 1).toLowerCase() === `${key}:`) {
```
Comment (KingNoosh) [inline mailheader.js]: Remember to update this to `header`.

File: tests/mailheader.js [hunk 3]
```diff
@@ -26,62 +26,75 @@ for (let i=0; i<lines.length; i++) {
...
-            h.get_decoded('content-type'),
+            this.h.get_decoded('content-type'),
             'multipart/alternative;   boundary=Apple-Mail-F2C5DAD3-7EB3-409D-9FE0-135C9FD43B69'
         );
-        test.equal(h.get_decoded('fromUTF8'), 'Kohl’s <Kohls@s.kohls.com>');
+        test.equal(this.h.get_decoded('fromUTF8'), 'Kohl’s <Kohls@s.kohls.com>');
         test.done();
     },
-    'content type w/parens': test => {
+    'content type w/parens': function (test) {
```
Comment (KingNoosh) [inline tests/mailheader.js]: What benefit does this change have?

Comment (msimerson) [reply tests/mailheader.js]: It allows `this` within the setup function to also refer to `this` within the test.

Comment (KingNoosh) [reply tests/mailheader.js]: Ah scoping, cool no problem.

File: mailheader.js (hunk 1 above)
Comment (msimerson) [reply mailheader.js]: If you notice, `i` gets used within the loop (in the splice). Kinda important...

Comment (KingNoosh) [reply mailheader.js]: I should probably look into getting new glasses.

**Pull request merged**